File: src/string/mostFrequentSymbol/TestMostFrequentSymbol.java
package string.mostFrequentSymbol;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

public class TestMostFrequentSymbol {
    @ParameterizedTest
    @CsvSource(value = {
            "aaaabbc, a",
            "abaabbcccc, c",
            "pppppppppppppppppppppuuuuuuuuuuuuuuuuuuuuuuqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq, q",
            "asasasasasppppppppppppppppppppppppppqw, p",
            "aaaAAAA, A",
            "привет питонннннннннннннннНННН, н",
            "11111111111111111111111111111111111111, 1",
            "jfnmdbsdfnsjfqenfdssjdfhsdjlkpppppppppppppppp"
                    + "ppgggggxxzzzssswwwwwwwwwwwwwwwwwwfgdfxdfg, w",
    })
    void testGetMostFrequentSymbol() {

    }
}

File: src/string/mostFrequentSymbol/test_most_frequent_symbol.py
from most_frequent_symbol import *
import pytest

data = [
    ('aaaabbc', 'a'),
    ('abaabbcccc', 'c'),
    ('pppppppppppppppppppppuuuuuuuuuuuuuuuuuuuuuuqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq', 'q'),
    ('asasasasasppppppppppppppppppppppppppqw', 'p'),
    ('aaaAAAA', 'A'),
    ('привет питонннннннннннннннНННН', 'н'),
    ('11111111111111111111111111111111111111', '1'),
    ("""jfnmdbsdfnsjfqenfdssjdfhsdjlkppppppppppppppppppggggg
 xxzzzssswwwwwwwwwwwwwwwwwwfgdfxdfg""", 'w')
]


@pytest.mark.parametrize('input, expected', data)
def test_most_frequent_symbol(input, expected):
    assert expected == most_frequent_symbol(input)
